api/src/pcapi/routes/backoffice/non_payment_notices/non_payment_notices_blueprint.py

```diff
@@ -0,0 +1,115 @@
+                offerers_models.Venue.id,
+                offerers_models.Venue.name,
+                offerers_models.Venue.publicName,
+                offerers_models.Venue.managingOffererId,
+            ),
+            sa_orm.joinedload(offerers_models.NonPaymentNotice.offerer).load_only(
+                offerers_models.Offerer.id,
+                offerers_models.Offerer.name,
+                offerers_models.Offerer.siren,
+                offerers_models.Offerer.postalCode,
+            ),
+            sa_orm.joinedload(offerers_models.NonPaymentNotice.batch).load_only(finance_models.CashflowBatch.label),
+        )
+        .order_by(sa.desc(offerers_models.NonPaymentNotice.dateCreated))
+    )
+
+    notices = notices_query.all()
+    return notices
+
+
+@non_payment_notices_blueprint.route("", methods=["GET"])
+def list_notices() -> utils.BackofficeResponse:
+    notices = _get_notices()
+
+    return render_template(
+        "non_payment_notices/list.html",
+        rows=notices,
+    )
+
+
+@non_payment_notices_blueprint.route("/create", methods=["GET"])
+@utils.permission_required(perm_models.Permissions.MANAGE_NON_PAYMENT_NOTICES)
+def get_create_non_payment_notice_form() -> utils.BackofficeResponse:
+    form = forms.CreateNonPaymentNoticeForm()
+
+    return render_template(
+        "components/turbo/modal_form.html",
+        form=form,
+        dst=url_for("backoffice_web.non_payment_notices.create_non_payment_notice"),
+        div_id="create-non-payment-notice",  # must be consistent with parameter passed to build_lazy_modal
+        title="Créer un avis d'impayé",
+        button_text="Créer l'avis",
+    )
+
+
+@non_payment_notices_blueprint.route("/create", methods=["POST"])
+@utils.permission_required(perm_models.Permissions.MANAGE_NON_PAYMENT_NOTICES)
+def create_non_payment_notice() -> utils.BackofficeResponse:
+    form = forms.CreateNonPaymentNoticeForm()
+
+    if not form.validate():
+        flash(utils.build_form_error_msg(form), "warning")
+        return redirect(url_for("backoffice_web.non_payment_notices.list_notices"), code=303)
+
+    try:
+        notice = offerers_models.NonPaymentNotice(
+            amount=form.amount.data,
+            dateReceived=form.date_received.data,
+            emitterEmail=form.emitter_email.data,
+            emitterName=form.emitter_name.data,
+            noticeType=offerers_models.NoticeType[form.notice_type.data],
+            offererId=int(form.offerer.data[0]) if form.offerer.data[0] else None,
+            reference=form.reference.data,
+            venueId=int(form.venue.data[0]) if form.venue.data[0] else None,
+        )
+        db.session.add(notice)
+        db.session.flush()
+        if notice.venue:
+            history_api.add_action(
+                history_models.ActionType.NON_PAYMENT_NOTICE_CREATED,
+                author=current_user,
+                venue=notice.venue,
```

Comment:
Peut-être aussi sur l'offerer, en changeant la condition au-dessus en `if notice.offerer or notice.venue:`
```suggestion
                offerer=notice.offerer,
                venue=notice.venue,
```